src/App.js

```diff
@@ -14,11 +14,145 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    isStarted: false,
+    goods: [...goodsFromServer],
+    isReversed: false,
+    sortTypeOf: '',
+    goodLength: 1,
+  }
+
+  toggleStart = () => {
+    this.setState(state => ({
+      isStarted: !state.isStarted,
+    }));
+  }
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  sortString = () => {
+    this.setState(() => ({
+      sortTypeOf: 'string',
+      isReversed: false,
+    }));
+  }
+
+  sortNumber = () => {
+    this.setState(() => ({
+      sortTypeOf: 'number',
+      isReversed: false,
+    }));
+  }
+
+  setGoodLength = (number) => {
+    this.setState(() => ({
+      goodLength: number,
+    }));
+  }
+
+  handleChange = (event) => {
+    this.setState(({ goodLength: event.target.value }));
+  }
+
+  reset = () => {
+    this.setState(() => ({
+      isReversed: false,
+      sortTypeOf: '',
+      goodLength: 1,
+    }));
+  }
+
+  render() {
+    const {
+      goods, isStarted, isReversed, sortTypeOf, goodLength,
+    } = this.state;
+
+    const lengthOptions = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+    const goodsList = [...goods].filter(el => el.length >= goodLength);
```

Comment:
Вынеси это из рендера в метод, а тут просто будешь делать
```
{
  this.getFilteredGoods().map()
}
```

Reply:
```suggestion
    const goodsList  = goods.filter(el => el.length >= goodLength);
```
filter не меняет исходный массив, не надо делать копию